adafruit_requests.py

```diff
@@ -357,10 +360,74 @@ def __init__(
         self._session_id = session_id
         self._last_response = None
 
+    def _build_boundary_data(self, files: dict):  # pylint: disable=too-many-locals
+        boundary_string = self._build_boundary_string()
+        content_length = 0
+        boundary_objects = []
+
+        for field_name, field_values in files.items():
+            file_name = field_values[0]
+            file_handle = field_values[1]
+
+            boundary_data = f"--{boundary_string}\r\n"
+            boundary_data += f'Content-Disposition: form-data; name="{field_name}"'
+            if file_name is not None:
+                boundary_data += f'; filename="{file_name}"'
+            boundary_data += "\r\n"
+            if len(field_values) >= 3:
+                file_content_type = field_values[2]
+                boundary_data += f"Content-Type: {file_content_type}\r\n"
+            if len(field_values) >= 4:
+                file_headers = field_values[3]
+                for file_header_key, file_header_value in file_headers.items():
+                    boundary_data += f"{file_header_key}: {file_header_value}\r\n"
+            boundary_data += "\r\n"
+
+            content_length += len(boundary_data)
+            boundary_objects.append(boundary_data)
+
+            if hasattr(file_handle, "read"):
+                is_binary = False
+                try:
+                    content = file_handle.read(1)
+                    is_binary = isinstance(content, bytes)
+                except UnicodeError:
+                    is_binary = False
```

Comment:
We could really use `file_handle.mode` here, hunh? No other way to find out whether the file is opened as text or binary.

Reply:
I didn't remember seeing a `mode`. Should it exist?

Reply:
It exists in CPython! Not in MicroPython/CircuitPython. I was just remarking here's a case where it would come in handy. Maybe it would be easy and small to add.